tests/test_templates_embeddings.py

```diff
@@ -186,24 +186,29 @@ def circuit(x=None):
 class TestAmplitudeEmbedding:
     """ Tests the pennylane.templates.embeddings.AmplitudeEmbedding method."""
 
-    def test_amplitude_embedding_state(self):
-        """Checks the state produced by pennylane.templates.embeddings.AmplitudeEmbedding()."""
+    @pytest.mark.parametrize("inpt", [np.array([0, 1, 0, 0]),
+                                      1/np.sqrt(4)*np.array([1, 1, 1, 1]),
+                                      np.array([np.complex(-np.sqrt(0.1), 0.0), np.sqrt(0.3),
+                                                np.complex(0, -np.sqrt(0.1)), np.sqrt(0.5)])])
+    def test_amplitude_embedding_prepares_state(self, inpt):
+        """Checks the state produced by pennylane.templates.embeddings.AmplitudeEmbedding() for real and complex
+        inputs."""
 
-        features = np.array([0, 1, 0, 0])
         n_qubits = 2
-        dev = qml.device('default.qubit', wires=n_qubits)
+        dev = qml.device('default.qubit', wires=2)
 
         @qml.qnode(dev)
         def circuit(x=None):
             AmplitudeEmbedding(features=x, wires=range(n_qubits), pad=False, normalize=False)
             return [qml.expval(qml.PauliZ(i)) for i in range(n_qubits)]
 
-        res = circuit(x=features)
-        assert np.allclose(res, [1, -1])
+        circuit(x=inpt)
```

Comment:
How come you pass as a keyword arg? Does it not work if the input is a regular argument?

Reply:
This is the test for a non-variable (i.e., a numpy array), since it gets passed via keyword argument. Below is the test for a list of Variable objects, which uses positional arguments.